plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/quantity-selector/frontend.ts

```diff
@@ -123,8 +110,31 @@ store< QuantitySelectorStore >(
 
 				return currentQuantity + step <= max;
 			},
+			get inputQuantity() {
+				const { productId } = getContext< Context >();
+
+				return (
+					addToCartWithOptionsStore.state.draftQuantity ??
+					( addToCartWithOptionsStore.state.quantity?.[ productId ] ||
+						0 )
+				);
+			},
 		},
 		actions: {
+			// Hold a draft value in state to allow a non-committed intermediate state of the input.
+			storeDraftValue: (
+				event: HTMLElementEvent< HTMLInputElement >
+			) => {
+				if (
+					isNaN( Number( event.target.value ) ) ||
+					event.target.value === ''
+				) {
+					addToCartWithOptionsStore.state.draftQuantity = '';
+				} else {
+					addToCartWithOptionsStore.state.draftQuantity =
+						Number( event.target.value ) || 0;
+				}
+			},
 			increaseQuantity: (
 				event: HTMLElementEvent< HTMLButtonElement >
```

Comment:
Is there a reason for the `draftQuantity` state to be in the `add-to-cart-with-options` store instead of `add-to-cart-with-options-quantity-selector`? I feel this value is specific to the quantity selector, and the parent form shouldn't need to know about it. This would also move us in the direction of decoupling the two stores so the blocks can eventually be used separately.